etl/data_helpers/misc.py

```diff
@@ -836,3 +861,425 @@ def humanize_number(number, sig_figs=2):
+
+        try:
+            metadata_rows = _extract_metadata_rows(
+                table[column_name].metadata, GSHEET_EXPORT_CONFIG["INCLUDED_METADATA_FIELDS"]
+            )
+
+            if metadata_rows:
+                # Create DataFrame with proper column headers
+                metadata_df = pd.DataFrame(metadata_rows, columns=["Property", "Value"])
+                clean_name = _sanitize_sheet_name(column_name)
+                metadata_dfs[f"metadata_{clean_name}"] = metadata_df
+
+        except Exception as e:
+            print(f"Warning: Could not process metadata for column '{column_name}': {e}")
+            continue
+
+    return metadata_dfs
+
+
+def export_table_to_gsheet(
+    table: Table,
+    sheet_title: str,
+    update_existing: bool = True,
+    folder_id: Optional[str] = None,
+    role: Literal["reader", "commenter", "writer"] = "reader",
+    general_access: Literal["anyone", "domain", "user"] = "anyone",
+    include_metadata: bool = True,
+    metadata_variables: Optional[List[str]] = None,
+) -> tuple[str, str]:
+    """Export a Table to Google Sheets with improved error handling and performance."""
+    import time
+
+    start_time = time.time()
+
+    try:
+        # Input validation
+        if not _validate_inputs(table, sheet_title):
+            return "", ""
+
+        # Sanitize sheet title
+        sheet_title = sheet_title.strip()[: GSHEET_EXPORT_CONFIG["MAX_TITLE_LENGTH"]]
+
+        # Early exit if credentials not available
+        if _should_skip_export():
+            print(
+                "Warning: Google API credentials not found or script is not running locally. Skipping Google Sheets export."
+            )
+            return "", ""
+
+        # Convert and validate data
+        df = _prepare_dataframe(table)
+
+        # Create or update sheet
+        sheet = _update_or_create_sheet(sheet_title, folder_id, df, update_existing)
+
+        # Write main data
+        _write_main_data(sheet, df)
+
+        # Add metadata if requested
+        if include_metadata:
+            _add_metadata_tabs(sheet, table, metadata_variables)
+
+        # Set permissions
+        _set_permissions(sheet.sheet_id, role, general_access)
+
+        duration = time.time() - start_time
+        print(f"Export completed in {duration:.2f} seconds")
+
+        return sheet.url, sheet.sheet_id
+
+    except Exception as e:
+        duration = time.time() - start_time
+        print(f"Export failed after {duration:.2f} seconds: {e}")
+        return "", ""
+
+
+@retry_on_network_error()
+def _create_google_sheet(sheet_title: str, folder_id: Optional[str]) -> GoogleSheet:
+    """Create Google Sheet with retry logic."""
+    if folder_id:
+        # Create sheet in the specified folder
+        return GoogleSheet.create_sheet(title=sheet_title, folder_id=folder_id)
+    else:
+        # Create sheet in root
+        return GoogleSheet.create_sheet(title=sheet_title)
+
+
+def _update_or_create_sheet(
```

Comment:
Couldn't this logic be part of GoogleSheet.create_sheet, to avoid having the code scattered between google and misc? Maybe it would be good to also clarify what "sheet" means (both here and in the google module). It's unclear to me whether it means "a google sheet file" or "a sheet inside a google sheet file".

Reply:
Good point, I'll use tab when it's a tab inside a sheet.